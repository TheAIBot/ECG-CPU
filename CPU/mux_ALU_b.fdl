dp muc_ALU_b(in b : ns(32);
			 in intermediate : ns(23);
			 in useIntermediate : ns(1);
			 out correctB : ns(32))
{
	always
	{
		correctB = useIntermediate ? (ns(32))intermediate : b;
	}
}
