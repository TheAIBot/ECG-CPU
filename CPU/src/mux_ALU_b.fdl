dp mux_ALU_b(in b : tc(32);
	     in intermediate : tc(23);
	     in useIntermediate : ns(1);
	     out correctB : tc(32)){
	always	{
		correctB = useIntermediate ? (tc(32))intermediate : b;
	}
}
