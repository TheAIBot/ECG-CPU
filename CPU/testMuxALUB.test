#!/bin/bash
name="testMuxALUB"
cpp -P $name".fdl" | fdlsim 2 > result.txt
java -jar testVerifier.jar $name".expected" result.txt
if [ $? = "0" ]; then
    echo "Success $name"
else
	echo "Failed $name"
fi
