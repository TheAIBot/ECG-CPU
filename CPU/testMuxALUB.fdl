#include "src/mux_ALU_b.fdl"

dp testMux(out b : ns(32);
   	   out intermediate : ns(23);
	   out useIntermediate : ns(1);
	   in correctB : ns(32)){

	
	always{$display($dec, "Cycle:", $cycle, ". Testing advanced ALU mux for the CPU. b = ", b , ", intermediate = ", intermediate , " useIntermediate = ", useIntermediate, ", correctB = ", correctB);}
	sfg test_UseIntermediate {b = 55; intermediate = 35; useIntermediate = 1;}
	sfg test_UseBGiven {b = 55; intermediate = 35; useIntermediate = 0;}
}


fsm mux_testbench(testMux){
    initial sUseIntermediate;
    state sUseBGiven;
    @sUseIntermediate (test_UseIntermediate) -> sUseBGiven;
    @sUseBGiven (test_UseBGiven) -> sUseIntermediate;
}

system myFirstSystem {
	mux_ALU_b(b, intermediate, useIntermediate, correctB);
	testMux(b, intermediate, useIntermediate, correctB);
}
