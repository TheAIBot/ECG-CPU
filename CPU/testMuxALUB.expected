Cycle:0. Testing advanced ALU mux for the CPU. b = 55, intermediate = 35 useIntermediate = 1, correctB = 35
Cycle:1. Testing advanced ALU mux for the CPU. b = 55, intermediate = 35 useIntermediate = 0, correctB = 55
